src/yunohost/app.py

```diff
@@ -908,77 +908,101 @@ def confirm_install(confirm):
     permission_add(app=app_instance_name, permission="main", sync_perm=False)
 
     # Execute the app install script
-    install_retcode = 1
+    install_failed = True
     try:
         install_retcode = hook_exec(
             os.path.join(extracted_app_folder, 'scripts/install'),
             args=args_list, env=env_dict
         )[0]
+        # "Common" app install failure : the script failed and returned exit code != 0
+        install_failed = (install_retcode != 0)
+        if install_failed:
+            error = m18n.n('app_install_script_failed')
+            logger.exception(error)
+            operation_logger.error(error)
+    # Script got manually interrupted ... N.B. : KeyboardInterrupt does not inherit from Exception
     except (KeyboardInterrupt, EOFError):
-        install_retcode = -1
-    except Exception:
+        error = m18n.n('operation_interrupted')
+        logger.exception(error)
+        operation_logger.error(error)
+    # Something wrong happened in Yunohost's code (most probably hook_exec)
+    except Exception as e :
         import traceback
-        logger.exception(m18n.n('unexpected_error', error=u"\n" + traceback.format_exc()))
+        error = m18n.n('unexpected_error', error=u"\n" + traceback.format_exc())
+        logger.exception(error)
+        operation_logger.error(error)
     finally:
+        # Whatever happened (install success or failure) we check if it broke the system
+        # and warn the user about it
         try:
             broke_the_system = False
             _assert_system_is_sane_for_app(manifest, "post")
         except Exception as e:
             broke_the_system = True
-            error_msg = operation_logger.error(str(e))
+            logger.exception(str(e))
+            operation_logger.error(str(e))
+
+        # If the install failed or broke the system, we remove it
+        if install_failed or broke_the_system:
```

Comment:
From an UX pov I think we should tell the user which case is selected, like : "we found out that this app broke the system" vs "the installation script failed".

It might be covered earlier with all the logger.exception but I'm not sure :s 

Or at least at the end with the exception raised?

Reply:
Yes it's covered by the earlier `logger.exception` ... (dunno what it really does compared to `logger.error` but anyway the message is displayed)

Reply:
(it's just a different level of importance of exceptions, like exception > error, but I don't think we really handle them differently)